### PR TITLE
feat: update right-click context menu

### PR DESCRIPTION
**Pull request by dogfrogfog**
https://github.com/pyroscope-io/pyroscope/issues/1235

https://user-images.githubusercontent.com/6951209/179028292-20ba6fe0-333f-4a09-bb65-c9cf47c4833a.mp4

### CODE REVIEW

**Comment by Rperry2174** [comment]
couple of small tweaks here as well:
I think right now the "right click" message is a little to overwhelming in the tooltip. Lets:

- make the right click logo a little smaller (let me know if you need a smaller svg)
- make the text a little bit grey so it stands out less
- move the text to be on one line (hopefully it fits, if not let me know and we can reword it a little bit) or potentially increase tooltip width a little bit


image

**Comment by eh-am** [comment]
Now that I am thinking, would be nice to be able to "unhighlight" at any point. Currently you only get a "unhighlight" after clicking on the exact same node.

Also we should come up with a better name, "unhighlight" is kinda weird. Maybe "Clear highlight"?
Also "similar nodes" is kinda confusing, what's a node? What if we call "highlight function"?

Another thing I noticed is a bug we already had, when the flamegraph changes we should clear the highlight. Similar to what we do here https://github.com/pyroscope-io/pyroscope/blob/main/packages/pyroscope-flamegraph/src/FlameGraph/FlameGraphRenderer.tsx#L210

https://user-images.githubusercontent.com/6951209/179028292-20ba6fe0-333f-4a09-bb65-c9cf47c4833a.mp4

File: packages/pyroscope-flamegraph/src/FlameGraph/FlameGraphComponent/index.tsx [hunk 1]
```diff
@@ -162,19 +172,56 @@ export default function FlameGraphComponent(props: FlamegraphProps) {
...
+      };
+
+      const HighlightSimilarNodesItem = () => {
+        const onClick = () => {
+          setActiveItem({ name: barName });
+        };
+
+        const actionName =
+          selectedItem.isJust && selectedItem.value === barName
+            ? 'Unhighlight'
```
Comment (Rperry2174) [inline packages/pyroscope-flamegraph/src/FlameGraph/FlameGraphComponent/index.tsx:203]:
```suggestion
            ? 'Clear highlight'
```

File: packages/pyroscope-flamegraph/src/FlameGraph/FlameGraphComponent/index.tsx [hunk 2]
```diff
@@ -162,19 +172,56 @@ export default function FlameGraphComponent(props: FlamegraphProps) {
...
+
+      const HighlightSimilarNodesItem = () => {
+        const onClick = () => {
+          setActiveItem({ name: barName });
+        };
+
+        const actionName =
+          selectedItem.isJust && selectedItem.value === barName
+            ? 'Clear highlight'
+            : 'Highlight';
```
Comment (Rperry2174) [inline packages/pyroscope-flamegraph/src/FlameGraph/FlameGraphComponent/index.tsx:204]:
```suggestion
            : 'Highlight similar nodes';
```

File: packages/pyroscope-flamegraph/src/FlameGraph/FlameGraphComponent/index.tsx [hunk 3]
```diff
@@ -162,19 +172,56 @@ export default function FlameGraphComponent(props: FlamegraphProps) {
...
+
+        return (
+          <MenuItem key="highlight-similar-nodes" onClick={onClick}>
+            <FontAwesomeIcon icon={faHighlighter} />
+            {actionName} similar nodes
```
Comment (Rperry2174) [inline packages/pyroscope-flamegraph/src/FlameGraph/FlameGraphComponent/index.tsx:209]:
```suggestion
            {actionName}
```

**Review by Rperry2174 — APPROVED** [review]
Looks great! Thanks @dogfrogfog

**Pull request merged**